File: IntegracaoPHP/estornoSplit.php
<?php
require("vendor/autoload.php");
$pagarme = new PagarMe\Client('ak_test_jbIXhrHXHOaUUNKtUVrkT9HGL60SSg');

$transaction = $pagarme->transactions()->create([
    'amount' => 10000,
    'payment_method' => 'credit_card',
    'card_holder_name' => 'teste de codigo',
    'card_cvv' => '123',
    'card_number' => '4242424242424242',
    'card_expiration_date' => '1220',
    'postback_url' => 'http://postback.url',
    'split_rules' => [
      [
      'liable' => 'true',
      'charge_processing_fee' => 'true',
      'percentage' => '50',
      'charge_remainder_fee' => 'true',
      'recipient_id' => 're_ck0zkjmuj00gln86dajtwrfws'
      ],
      [
      'liable' => 'false',
      'charge_processing_fee' => 'false',
      'percentage' => '50',
      'charge_remainder_fee' => 'false',
      'recipient_id' => 're_ck0zjmw1s007ier6eysyd3agj'
      ]
      ],
    'customer' => [
        'external_id' => '1',
        'name' => 'Nome do cliente',
        'type' => 'individual',
        'country' => 'br',
        'documents' => [
          [
            'type' => 'cpf',
            'number' => '67415765095'
          ]
        ],
        'phone_numbers' => [ '+551199999999' ],
        'email' => 'cliente@email.com'
    ],
    'billing' => [
        'name' => 'Nome do pagador',
        'address' => [
          'country' => 'br',
          'street' => 'Avenida Brigadeiro Faria Lima',
          'street_number' => '1811',
          'state' => 'sp',
          'city' => 'Sao Paulo',
          'neighborhood' => 'Jardim Paulistano',
          'zipcode' => '01451001'
        ]
    ],
    'shipping' => [
        'name' => 'Nome de quem receberá o produto',
        'fee' => 1020,
        'delivery_date' => '2018-09-22',
        'expedited' => false,
        'address' => [
          'country' => 'br',
          'street' => 'Avenida Brigadeiro Faria Lima',
          'street_number' => '1811',
          'state' => 'sp',
          'city' => 'Sao Paulo',
          'neighborhood' => 'Jardim Paulistano',
          'zipcode' => '01451001'
        ]
    ],
    'items' => [
        [
          'id' => '1',
          'title' => 'R2D2',
          'unit_price' => 300,
          'quantity' => 1,
          'tangible' => true
        ],
        [
          'id' => '2',
          'title' => 'C-3PO',
          'unit_price' => 700,
          'quantity' => 1,
          'tangible' => true
        ]
    ]
  ]);
 $id_trx = strval($transaction -> id);
 $id_split_rules1 = strval($transaction -> split_rules[0] -> id);
 $id_split_rules2 = strval($transaction -> split_rules[1] -> id);
 sleep(5);

$refundedTransactionWithSplit = $pagarme->transactions()->refund([
    'id' => $id_trx,
    'amount' => 10000,
    'split_rules' => [
        [
            'id' => $id_split_rules2,
            'percentage' => '50',
            'recipient_id' => 're_ck0zkjmuj00gln86dajtwrfws',
            'charge_processing_fee' => 'true',
            'liable' => 'true',
            'charge_remainder_fee' => 'true'
        ],
        [
            'id' => $id_split_rules1,
            'percentage' => '50',
            'recipient_id' => 're_ck0zjmw1s007ier6eysyd3agj',
            'charge_processing_fee' => 'false',
            'liable' => 'false',
            'charge_remainder_fee' => 'false'
        ]
    ]
]);
echo("A transação número ".$refundedTransactionWithSplit->id." foi criada e estornada!");

File: IntegracaoPython/estornoSplit.py
import pagarme
import time
import json


def EstornoSplit():
    
  pagarme.authentication_key("ak_test_jbIXhrHXHOaUUNKtUVrkT9HGL60SSg")


  params = {
    "amount": "21000",
      "card_number": "4111111111111111",
      "card_cvv": "123",
      "card_expiration_date": "0922",
      "card_holder_name": "teste de codigo",
      "postback_url": "http://requestb.in/pkt7pgpk",
      "async": "false",
      "split_rules": [
          { 
          "recipient_id": "re_ck0zkjmuj00gln86dajtwrfws", 
          "percentage": "50",
          "liable": "false",
          "charge_processing_fee": "false"
          },
          { 
          "recipient_id": "re_ck0zjmw1s007ier6eysyd3agj", 
          "percentage": "50",
          "liable": "true",
          "charge_processing_fee": "true"
          }
      ],
      "customer": {
        "external_id": "#3311",
        "name": "Morpheus Fishburne",
        "type": "individual",
        "country": "br",
        "email": "mopheus@nabucodonozor.com",
        "documents": [
          {
            "type": "cpf",
            "number": "30621143049"
          }
        ],
        "phone_numbers": ["+5511999998888", "+5511888889999"],
        "birthday": "1965-01-01"
      },
      "billing": {
        "name": "Trinity Moss",
        "address": {
          "country": "br",
          "state": "sp",
          "city": "Cotia",
          "neighborhood": "Rio Cotia",
          "street": "Rua Matrix",
          "street_number": "9999",
          "zipcode": "06714360"
        }
      },
      "shipping": {
        "name": "Neo Reeves",
        "fee": "1000",
        "delivery_date": "2000-12-21",
        "expedited": True,
        "address": {
          "country": "br",
          "state": "sp",
          "city": "Cotia",
          "neighborhood": "Rio Cotia",
          "street": "Rua Matrix",
          "street_number": "9999",
          "zipcode": "06714360"
        }
      },
      "items": [
        {
          "id": "r123",
          "title": "Red pill",
          "unit_price": "10000",
          "quantity": "1",
          "tangible": True
        },
        {
          "id": "b123",
          "title": "Blue pill",
          "unit_price": "10000",
          "quantity": "1",
          "tangible": True
        }
      ]
  }

  trx = pagarme.transaction.create(params)

  id_trx = str(trx['id'])
  time.sleep(5)

  transaction = pagarme.transaction.find_by({'id': id_trx})

  estr = pagarme.transaction.refund(transaction[0]['id'], {"amount": "21000"})
  
  estr_in_string = json.dumps(estr)
  parsed_json = (json.loads(estr_in_string))
  
  return parsed_json["id"]

